vault/expiration.go

```diff
@@ -407,6 +445,29 @@ func (m *ExpirationManager) Restore(errorFunc func()) (retErr error) {
 	m.restoreModeLock.Unlock()
 
 	m.logger.Info("lease restore complete")
+
+	// At this time all the leases are loaded. The restoration of leases would
+	// have also computed if were any leases that had either its ID or
+	// secondary index obfuscated using SHA1 hash instead of SHA2-256 HMAC.
+	// Persist the result of that computation across reboots.
+	expMetadata := &ExpirationMetadata{}
+
+	if atomic.LoadInt32(&m.SHA1HashedLeasesPresent) == 0 {
```

Comment:
This can (and should) all be skipped if `m.SHA1HashedLeasesCleared` is already set to true

Reply:
This is the right thing to do. Fixed. This will fail the test as the seal/unseal cycles during startup will set the `Cleared` var and if the test tries to insert an older lease entry, it won't reflect the state. But it means that the test should be fixed. So 👍